webapp/tests/karma/ts/services/select2-search.service.spec.ts

```diff
@@ -0,0 +1,136 @@
+        { provide: LineageModelGeneratorService, useValue: lineageModelGeneratorService },
+        { provide: SearchService, useValue: { } },
+        { provide: SessionService, useValue: sessionService },
+        { provide: SettingsService, useValue: settingsService },
+      ]
+    });
+    service = TestBed.inject(Select2SearchService);
+  });
+
+  afterEach(() => sinon.restore());
+
+  describe('init', () => {
+    it('should set empty value when initial value is empty', async () => {
+      await service.init(selectEl, [ 'person' ], {initialValue: ''});
+      expect(selectEl.val.callCount).to.equal(3);
+      expect(selectEl.val.args[0]).to.deep.equal([ ]);    // first time the component reads the current value
+      expect(selectEl.val.args[1]).to.deep.equal([ '' ]); // set the value
+      expect(selectEl.val.args[2]).to.deep.equal([ ]);    // read the value
+      expect(val).to.equal('');                           // current value ''
+      expect(lineageModelGeneratorService.contact.callCount).to.equal(0);   // empty initial value => no calls to DB
+      expect(selectEl.trigger.args[0]).to.deep.equal([ 'change' ]);         // the change is notified to the component
+    });
+
+    it('should set right value when a valid reference is set', async () => {
+      selectEl.children.returns({ length: 0 });
+      const person = {
+        _id: 'aaa-222-333',
+        doc: {
+          _id: 'aaa-222-333',
+          _rev: '1-000123',
+          type: 'person',
+          name: 'Charles C'
+        },
+      };
+      lineageModelGeneratorService.contact.resolves(person);
+      await service.init(selectEl, [ 'person' ], { initialValue: 'aaa-222-333' });
+      expect(selectEl.val.callCount).to.equal(2);
+      expect(selectEl.val.args[0]).to.deep.equal([ 'aaa-222-333' ]); // set the value
+      expect(selectEl.val.args[1]).to.deep.equal([ ]);               // read the value
+      expect(lineageModelGeneratorService.contact.callCount).to.equal(1);
+      expect(lineageModelGeneratorService.contact.args[0]).to.deep.equal([ 'aaa-222-333', { merge: true } ]);
+      expect(contactMutedService.getMuted.callCount).to.equal(1);
+      expect(contactMutedService.getMuted.args[0]).to.deep.equal([ person.doc ]);
+      expect(val).to.equal('aaa-222-333');                           // final value
+      expect(select2Val[0]).to.deep.equal({ doc: person.doc });
+      expect(selectEl.trigger.callCount).to.equal(1);
+      expect(selectEl.trigger.args[0]).to.deep.equal([ 'change' ]);  // the change is notified to the component
+    });
+
+    it('should set the id value when a deleted reference is set and fixed label', async () => {
+      selectEl.children.returns({ length: 0 });
+      lineageModelGeneratorService.contact.rejects({ code: 404, error: 'not found' });
+      try {
+        await service.init(selectEl, ['person'], {initialValue: 'aaa-222-333'});
+        assert.fail('expected error to be thrown');
+      } catch (e) {
```

Comment:
I don't think we expect an error here, we actually expect no error! 
The catch is actually catching the error you throw with `assert.fail('expected error to be thrown');`.

Reply:
Wrong copy & paste :smile:

Reply:
I think this has not been addressed?

Reply:
I did, at https://github.com/medic/cht-core/pull/7158/commits/7b56e0981620f1c4414a5a551d2f65621697cd79, but like you I'm don't see the "Outdated" bubble that Github usually use to tag the outdated suggestions, but if you checkout the "Files changed" tab you will also note the change.

Reply:
I refreshed the files changed lots of times and: 
![image](https://user-images.githubusercontent.com/35681649/145405171-2c58c8b4-66e3-4716-93d7-bf7e86dad86c.png)

The change I requested was to remove the try-catch entirely.

Reply:
:facepalm: my bad, I misunderstood your suggestion, you are right... pushed the fix.

Reply:
@dianabarsan , I have removed the try-catch , sorry for the misunderstanding of your request, please review again :pray: